vector/src/main/java/im/vector/app/features/matrixto/OriginOfMatrixTo.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.matrixto
+
+import im.vector.app.features.analytics.plan.JoinedRoom
+import im.vector.app.features.analytics.plan.ViewRoom
+
+enum class OriginOfMatrixTo {
+    LINK,
+    NOTIFICATION,
+    TIMELINE,
+    SPACE_EXPLORE,
+    ROOM_LIST,
+    USER_CODE
+}
+
+fun OriginOfMatrixTo.toJoinedRoomTrigger(): JoinedRoom.Trigger {
+    return when (this) {
+        OriginOfMatrixTo.LINK          -> JoinedRoom.Trigger.MobilePermalink
+        OriginOfMatrixTo.NOTIFICATION  -> JoinedRoom.Trigger.Notification
+        OriginOfMatrixTo.TIMELINE      -> JoinedRoom.Trigger.Timeline
+        OriginOfMatrixTo.SPACE_EXPLORE -> JoinedRoom.Trigger.MobileExploreRooms
+        OriginOfMatrixTo.ROOM_LIST     -> JoinedRoom.Trigger.RoomDirectory
+        OriginOfMatrixTo.USER_CODE     -> throw IllegalArgumentException("can't map source to join room trigger")
+    }
+}
+
+fun OriginOfMatrixTo.toViewRoomTrigger(): ViewRoom.Trigger {
+    return when (this) {
+        OriginOfMatrixTo.LINK          -> ViewRoom.Trigger.MobilePermalink
+        OriginOfMatrixTo.NOTIFICATION  -> ViewRoom.Trigger.Notification
+        OriginOfMatrixTo.TIMELINE      -> ViewRoom.Trigger.Timeline
+        OriginOfMatrixTo.SPACE_EXPLORE -> ViewRoom.Trigger.MobileExploreRooms
+        OriginOfMatrixTo.ROOM_LIST     -> ViewRoom.Trigger.RoomDirectory
+        OriginOfMatrixTo.USER_CODE     -> throw IllegalArgumentException("can't map source to view room trigger")

```

Comment:
I did not see that at first review, but this may crash the app. Also at line 38. Can't we just return `null` in this case to avoid any potential crash?
The 2 methods can return nullable, hence the caller can decide to send the analytics without the info, or not send the analytics.